### PR TITLE
WLA-456: Letter for Grant issued state and Letter for Grant issued - …

### PR DESCRIPTION
**Pull request by rishikrsharma**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/WLA-456
https://tools.hmcts.net/jira/browse/WLA-457

### Change description ###
[BO] Letter for Grant issued - Intestacy
[BO] Letter for Grant issued state - Probate case

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/probate/controller/DocumentController.java [hunk 1]
```diff
@@ -160,11 +160,13 @@
         Document coverSheet = pdfManagementService.generateAndUpload(callbackRequest, DocumentType.GRANT_COVER);
         log.info("Generated and Uploaded cover document with template {} for the case id {}",
                 DocumentType.GRANT_COVER.getTemplateName(), callbackRequest.getCaseDetails().getId().toString());
-
+        Document letterOfGrantIssuedState = null;
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/controller/DocumentController.java]: why did you call it letterOfGrantIssuedState? isnt it just letterOfGrantIssued? and the other type is letterOfGrantIssuedIntestacy. For Doc Types too

File: src/main/java/uk/gov/hmcts/probate/controller/DocumentController.java [hunk 2]
```diff
@@ -179,7 +181,9 @@
         List<Document> documents = new ArrayList<>();
         documents.add(digitalGrantDocument);
         documents.add(coverSheet);
-
+        if(letterOfGrantIssuedState != null){
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/controller/DocumentController.java]: The ordering is wrong according to the JIRA

File: src/main/java/uk/gov/hmcts/probate/controller/NotificationController.java [hunk 3]
```diff
@@ -144,7 +144,7 @@
                 log.info("Initiate call to bulk print for Caveat stopped document and coversheet for case id {} ",
                         callbackRequest.getCaseDetails().getId());
                 SendLetterResponse sendLetterResponse =
-                    bulkPrintService.sendToBulkPrintForGrant(callbackRequest, caveatRaisedDoc, coversheet);
+                    bulkPrintService.sendToBulkPrintForGrant(callbackRequest, caveatRaisedDoc, null, coversheet);
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/controller/NotificationController.java]: Why dont you override the bulkPrintService method with sendToBulkPrintForLetterOfGrantIssued(callbackRequest, caveatRaisedDoc, coversheet)

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -99,6 +99,10 @@ docmosis:
       templateName: FL-PRB-GRP-WEL-00226.doc
     welshAdmonWillGrantDraft:
       templateName: FL-PRB-GRP-WEL-00226.doc
+    letterOfGrantIssuedState:
```
Comment (sanjayparekh-hmcts) [inline src/main/resources/application.yml]: same name change as above here please

File: src/test/java/uk/gov/hmcts/probate/controller/DocumentControllerTest.java [hunk 5]
```diff
@@ -643,7 +654,36 @@ public void shouldSendForReprint() throws Exception {
...
+        when(documentGeneratorService.generateLetterOfGrantDelay(any(CallbackRequest.class),
+                eq(DocumentType.LETTER_OF_GRANT_ISSUED_INTESTACY)))
+                .thenReturn(letter);
+        String solicitorPayload = testUtils.getStringFromFile("solicitorPayloadNotificationsBulkPrintIntestacy.json");
+
+        MvcResult result = mockMvc.perform(post("/document/generate-grant")
+                .content(solicitorPayload)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.data.probateDocumentsGenerated[1].value.DocumentType",
```
Comment (sanjayparekh-hmcts) [inline src/test/java/uk/gov/hmcts/probate/controller/DocumentControllerTest.java:680]: Test order of docs here?

**Review by sanjayparekh-hmcts — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/probate/controller/DocumentController.java (hunk 2 above)
Comment (rishikrsharma) [reply src/main/java/uk/gov/hmcts/probate/controller/DocumentController.java]: This is not actual order, actual order of print has been re-arranged in bulk print service. Originally we were adding grant then cover.

**Review by sanjayparekh-hmcts — APPROVED** [review]

**Pull request closed without merging**